app/models/project_metadata.rb

```diff
@@ -1,125 +1,120 @@
-
-    # activate a project by setting the status to 'active'
-    project.metadata_json["status"] = Project::ACTIVE_STATUS
-
-    # also read in the actual project directory
-    project.metadata_json["project_directory"] = metadata[:project_directory]
-
-    project.save!
-
-    # create two provenance events, one for approving the project and another for changing the status of the project
-    project.provenance_events.create(event_type: ProvenanceEvent::ACTIVE_EVENT_TYPE, event_person: current_user.uid, event_details: "Activated by Tigerdata Staff")
-    project.provenance_events.create(event_type: ProvenanceEvent::STATUS_UPDATE_EVENT_TYPE, event_person: current_user.uid, event_details: "The Status of this project has been set to active")
-  end
-
-  # Approve a project by recording the mediaflux id & setting the status to 'approved'
-  # @param params [Hash] the updated Project metadata attributes
-  def approve_project(params:)
-    project.mediaflux_id = params[:mediaflux_id]
-    project.metadata_json["status"] = Project::APPROVED_STATUS
-    project.metadata_json["project_directory"] = "#{params[:project_directory_prefix]}/#{params[:project_directory]}"
-    project.metadata_json["storage_capacity"] = params[:storage_capacity]
-    project.metadata_json["storage_performance_expectations"] = params[:storage_performance_expectations]
-
-    project.save!
-    generate_approval_events(params[:approval_note])
-  end
-
-  def generate_approval_events(note)
-    # create two provenance events, one for approving the project and another for changing the status of the project
-    project.provenance_events.create(event_type: ProvenanceEvent::APPROVAL_EVENT_TYPE, event_person: current_user.uid, event_details: "Approved by #{current_user.display_name_safe}",
-                                     event_note: note)
-    project.provenance_events.create(event_type: ProvenanceEvent::STATUS_UPDATE_EVENT_TYPE, event_person: current_user.uid, event_details: "The Status of this project has been set to approved")
+  # rubocop:disable Metrics/MethodLength
+  def initialize_from_hash(metadata_hash)
+    @title = metadata_hash[:title]
+    @description = metadata_hash[:description]
+    @status = metadata_hash[:status]
+    @data_sponsor = metadata_hash[:data_sponsor]
+    @data_manager = metadata_hash[:data_manager]
+    @departments = metadata_hash[:departments]
+    @ro_users = metadata_hash[:ro_users]
+    @rw_users = metadata_hash[:rw_users]
+
+    @project_id = metadata_hash[:project_id]
+    @project_purpose = metadata_hash[:project_purpose]
+    @project_directory = metadata_hash[:project_directory]
+
+    @storage_capacity = metadata_hash[:storage_capacity]
+    @storage_performance_expectations = metadata_hash[:storage_performance_expectations]
+    @created_on = metadata_hash[:created_on]
+    @created_by = metadata_hash[:created_by]
+    @updated_by = metadata_hash[:updated_by]
+    @updated_on = metadata_hash[:updated_on]
+    set_defaults
```

Comment:
Using this version of the defaults will cause a regression.  We need to be utilizing the `Rails.configuration.project_defaults` here

Reply:
Good catch, we are not reading them from the config again